src/public/server.ts

```diff
@@ -23,13 +28,26 @@ export function getPublicServer(
   httpServer: Server
 ): ApolloServer<IPublicContext> {
   const cache = getRedisCache();
+
+  // Add @constraint directive to the schema
+  const schema = buildSubgraphSchema([
+    { typeDefs: gql(constraintDirectiveTypeDefs) },
+    {
+      typeDefs: typeDefsPublic,
+      resolvers,
```

Comment:
Not 100% sure, but if the link directive stuff from slack doesn't solve the compilation issue all the way, try compiling typedefs into one array

```suggestion
    { typeDefs: [gql(constraintDirectiveTypeDefs), typeDefsPublic],
      resolvers,
```